test/basic.test.js

```diff
@@ -58,8 +57,179 @@ test('Subscribed clients receive updates', async function (t) {
   var publisher = await helper.startClient()
 
   await publisher.publish(msg.topic, msg.payload, msg)
-  await helper.delay(2000)
+  await helper.delay(500)
 
   await Promise.all(subscribers.map(c => c.end()))
   await publisher.end()
+}
+
+test('Subscribed clients receive updates - QoS 1', async function (t) {
+  t.plan(10, 'each client should receive a message')
+  t.tearDown(helper.closeBroker)
+
+  await helper.startBroker()
+
+  await testQos(t, 1)
+})
+
+test('Subscribed clients receive updates - QoS 2', async function (t) {
+  t.plan(10, 'each client should receive a message')
+  t.tearDown(helper.closeBroker)
+
+  await helper.startBroker()
+  await testQos(t, 2)
+})
+
+test('Connect clean=false', async function (t) {
+  t.plan(1)
+  t.tearDown(helper.closeBroker)
+
+  const options = { clientId: 'pippo', clean: false }
+
+  await helper.startBroker()
+
+  var publisher = await helper.startClient('mqtt', options)
+
+  await publisher.subscribe('my/topic')
+
+  await publisher.end(true)
+
+  publisher = await helper.startClient('mqtt', options)
+
+  publisher.once('message', function (topic, message) {
+    if (topic === 'my/topic') {
+      t.pass('Subscription has been restored')
+    }
+  })
+
+  await publisher.publish('my/topic', 'I\'m alive', { qos: 1 })
+
+  await helper.delay(500)
```

Comment:
This is not needed. You can await a promise that will be resolved inside the `once('message')` handler.